### PR TITLE
Extract World SNO ids into enum and replace int "magic constants"

### PR DESCRIPTION
**Pull request by DeKaN**
The main goal is rid away from using "magic constants". This PR touches only World SNO ids

### CODE REVIEW

File: src/DiIiS-NA/D3-GameServer/GSSystem/MapSystem/World.cs [hunk 1]
```diff
@@ -70,6 +71,10 @@ public sealed class World : DynamicObject, IRevealable, IUpdateable
 		/// SNOHandle for the world.
 		/// </summary>
 		public SNOHandle WorldSNO { get; private set; }
```
Comment (DeKaN) [inline src/DiIiS-NA/D3-GameServer/GSSystem/MapSystem/World.cs:73]: Usage outside this class has been minimized. Can be made private

File: src/DiIiS-NA/D3-GameServer/GSSystem/PowerSystem/Implementations/MonsterSkills/SingleProjectileSkill.cs [hunk 2]
```diff
@@ -31,6 +31,7 @@ public abstract class SingleProjectileSkill : ActionTimedSkill
 		protected void SetProjectile(PowerContext context, int actorSNO, Vector3D position, float speed = 1f, Action<Actor> OnCollision = null)
 		{
 			if (User is Monster)
+				// FIXME: Non-exist world id
 				if (User.World.WorldSNO.Id == 1 ||
```
Comment (DeKaN) [inline src/DiIiS-NA/D3-GameServer/GSSystem/PowerSystem/Implementations/MonsterSkills/SingleProjectileSkill.cs:35]: Found strange code, just to bring attention to it

File: src/DiIiS-NA/D3-GameServer/GSSystem/QuestSystem/ActI.cs [hunk 3]
```diff
@@ -277,14 +282,15 @@ public override void SetQuests()
                     UnlockTeleport(1);
                     ListenTeleport(101351, new Advance());
                     //AddFollower(this.Game.GetWorld(71150), 4580);
-                    this.Game.AddOnLoadAction(71150, () =>
```
Comment (DeKaN) [inline src/DiIiS-NA/D3-GameServer/GSSystem/QuestSystem/ActI.cs:280]: Possible code duplication, please check

File: src/DiIiS-NA/D3-GameServer/GSSystem/QuestSystem/ActII.cs [hunk 4]
```diff
@@ -594,17 +606,19 @@ public override void SetQuests()
...
 					{
 						if (this.Game.CurrentQuest == 57331 && this.Game.CurrentStep == 31)
 						{
-							this.Game.GetWorld(86594).ShowOnlyNumNPC(3205, -1); //Leave all Asheara
-							this.Game.GetWorld(86594).ShowOnlyNumNPC(4580, -1); //Leave all Leah
-							this.Game.GetWorld(86594).ShowOnlyNumNPC(4580, -1); //Leave all Adria
+							var world = this.Game.GetWorld(WorldSno.a2dun_cald);
+							world.ShowOnlyNumNPC(3205, -1); //Leave all Asheara
+							world.ShowOnlyNumNPC(4580, -1); //Leave all Leah
+							// FIXME: check NPC snoID, maybe it should be 3095
```
Comment (DeKaN) [inline src/DiIiS-NA/D3-GameServer/GSSystem/QuestSystem/ActII.cs:616]: Attempt for using one Actor SnoId as two different actors, please check

File: src/DiIiS-NA/D3-GameServer/GSSystem/QuestSystem/ActIII.cs [hunk 5]
```diff
@@ -214,17 +221,19 @@ public override void SetQuests()
...
 					UnlockTeleport(3);
-					this.Game.AddOnLoadAction(103209, () =>
+					this.Game.AddOnLoadWorldAction(WorldSno.gluttony_boss, () =>
 					{
 						if (this.Game.CurrentQuest == 93697)
 						{
-							ActII.DisableEveryone(this.Game.GetWorld(103209), true);
+							var world = this.Game.GetWorld(WorldSno.gluttony_boss);
+							// TODO: extract this static method as extension
+							ActII.DisableEveryone(world, true);
```
Comment (DeKaN) [inline src/DiIiS-NA/D3-GameServer/GSSystem/QuestSystem/ActIII.cs:230]: I think this static method can be extracted from `ActII` class

File: src/DiIiS-NA/D3-GameServer/GSSystem/QuestSystem/ActV.cs [hunk 6]
```diff
@@ -310,26 +319,28 @@ public override void SetQuests()
...
+                            ActiveArrow(this.Game.GetWorld(WorldSno.x1_westmarch_hub), 315793);
 					});
-					//Enter Westmarch Commons
-					ListenTeleport(261758, new BackToCath());
-					foreach (var Myst in this.Game.GetWorld(338891).GetActorsBySNO(339463)) //Mystic
+                    //Enter Westmarch Commons
+                    ListenTeleport(261758, new BackToCath());
+                    var world = this.Game.GetWorld(WorldSno.x1_westm_deathorb_gideonscourt);
+					// FIXME: incorrect snoId or possible code duplicate
+                    foreach (var Myst in world.GetActorsBySNO(339463)) //Mystic
```
Comment (DeKaN) [inline src/DiIiS-NA/D3-GameServer/GSSystem/QuestSystem/ActV.cs:338]: Using `339463` for actors `Mystic` and `Mystic_EnchanceEvent`. Possible mistake

**Comment by DeKaN** [comment]
@iEvengel @pr701 Please review :)

**Review by iEvengel — APPROVED** [review]

**Pull request merged**